### PR TITLE
chore(DATAGO-123909): Create inline citation and sources chip

### PR DESCRIPTION
**Pull request by JKaram**
### What is the purpose of this change?

Enhance the citation UI to display more informative inline citations with page positions, and add a sources chip for document search results.

**Key improvements:**
- Document citations now show filename with page position (e.g., "report.pdf, Pages 3-5")
- Bundled citations show "+N" indicator when multiple sources are grouped
- Sources chip now appears for document search results (previously only web/deep research)
- Added Storybook stories for Citation components

### How was this change implemented?

**Frontend changes:**
- `Citation.tsx` - Enhanced display text to include page position from metadata, increased max display length, removed tooltip for document citations (they open RAG panel instead)
- `ChatMessage.tsx` - Added `document_search` type to sources chip visibility condition
- `citations.ts` - Refactored `parseCitationId` to handle all citation types (search, document, research)
- Added Storybook stories and unit tests for Citation and BundledCitations components

### Key Design Decisions

- **No popover for document citations**: Document citations open the RAG info panel on click, so a hover popover would be redundant. Only web/research citations show popovers since they link to external URLs.
- **Page position in badge**: Shows location info directly in the badge for quick context without needing to click.

### How was this change tested?

- [x] Manual testing: Verified citations display correctly in chat with document search results
- [x] Unit tests: Added edge case tests for empty citations and deduplication
- [x] Storybook tests: 9 visual stories covering all citation types and bundling scenarios

### Screenshots

**Chat with document citations showing filename + page position:**

**Closeup of citation badges:**

### Is there anything the reviewers should focus on?

- The citation display changes are refinements to existing functionality, not a new feature
- No feature flag needed for frontend - changes are backward compatible and enhance existing citation rendering

image
image

### CODE REVIEW

File: client/webui/frontend/src/lib/components/chat/ChatMessage.tsx [hunk 1]
```diff
@@ -990,7 +990,10 @@ export const ChatMessage: React.FC<{ message: MessageFE; isLastWithTaskId?: bool
     // Check if this is a completed web search message (has web_search sources but not deep research)
     const isWebSearchComplete = message.isComplete && !isDeepResearchComplete && hasRagSources && taskRagData?.some(r => r.searchType === "web_search");
 
-    // Handler for sources click (works for both deep research and web search)
+    // Check if this is a completed document search message (has document_search sources)
+    const isDocumentSearchComplete = message.isComplete && !isDeepResearchComplete && hasRagSources && taskRagData?.some(r => r.searchType === "document_search");
```
Comment (lgh-solace) [inline client/webui/frontend/src/lib/components/chat/ChatMessage.tsx]: Should we extract `message.isComplete && !isDeepResearchComplete && hasRagSources ` to common const so we don't have to recheck?

Comment (lgh-solace) [reply client/webui/frontend/src/lib/components/chat/ChatMessage.tsx]: Also, how are document sources feature flagged? Will the BE not return these sources if project indexing is disabled?

File: client/webui/frontend/src/lib/components/chat/ChatSidePanel.tsx [hunk 2]
```diff
@@ -28,8 +28,17 @@ export const ChatSidePanel: React.FC<ChatSidePanelProps> = ({ onCollapsedToggle,
...
+    // Check if there are any sources in the current session
+    // Includes: web sources, deep research sources, AND document search sources
+    const hasSourcesInSession = useMemo(() => {
+        if (!ragData || ragData.length === 0) return false;
+        
+        // Check for web/research sources with URLs (existing behavior)
+        if (hasSourcesWithUrls(ragData)) return true;
+        
+        // Also check for document search results
+        return ragData.some(search => search.searchType === "document_search");
```
Comment (lgh-solace) [inline client/webui/frontend/src/lib/components/chat/ChatSidePanel.tsx]: I wonder if we could have a utility for this check?

File: client/webui/frontend/src/lib/components/chat/Citation.tsx [hunk 3]
```diff
@@ -276,22 +298,38 @@ export function BundledCitations({ citations, onCitationClick }: BundledCitation
...
+        // For document citations, use onClick handler (to open RAG panel)
         if (onCitationClick) {
             onCitationClick(firstCitation);
         }
     };
 
+    // For document citations: render simple button without popover
+    if (isDocumentCitation) {
+        return (
+            <button
```
Comment (lgh-solace) [inline client/webui/frontend/src/lib/components/chat/Citation.tsx:310]: Could this be a standard `Button`? If not, can we use theme colours instead of the tailwind grays?

File: client/webui/frontend/src/lib/components/chat/Citation.tsx [hunk 4]
```diff
@@ -276,22 +298,38 @@ export function BundledCitations({ citations, onCitationClick }: BundledCitation
...
+    // For web/research citations: render with popover
     return (
         <Popover open={isOpen} onOpenChange={setIsOpen}>
             <PopoverTrigger asChild>
                 <button
-                    onClick={handleFirstCitationClick}
+                    onClick={handleClick}
                     onMouseEnter={handleMouseEnter}
                     onMouseLeave={handleMouseLeave}
                     className="citation-badge mx-0.5 inline-flex cursor-pointer items-center gap-1 rounded-sm bg-gray-200 px-1.5 py-0 align-baseline text-[11px] font-normal whitespace-nowrap text-gray-800 transition-colors duration-150 hover:bg-gray-300 dark:bg-gray-700 dark:text-white dark:hover:bg-gray-600"
```
Comment (lgh-solace) [inline client/webui/frontend/src/lib/components/chat/Citation.tsx]: I guess the styles were from here... if you have time to update them?

File: client/webui/frontend/src/lib/components/web/StackedFavicons.tsx [hunk 5]
```diff
@@ -7,10 +7,11 @@
 import { getCleanDomain, getFaviconUrl } from "@/lib/utils";
 
 interface SearchSource {
```
Comment (lgh-solace) [inline client/webui/frontend/src/lib/components/web/StackedFavicons.tsx]: Two interfaces for `SearchSource`, can we consolidate to one?

File: client/webui/frontend/src/lib/components/web/StackedFavicons.tsx [hunk 6]
```diff
@@ -21,11 +22,60 @@ interface StackedFaviconsProps {
 }
 
 /**
- * Source icon component - handles both web favicons and enterprise source icons
+ * Get file type icon emoji based on file extension
+ * Supports: PDF, DOCX, PPTX, and text-based files (txt, md, csv, json, xml, etc.)
+ */
+function getFileTypeIcon(filename: string): string {
```
Comment (lgh-solace) [inline client/webui/frontend/src/lib/components/web/StackedFavicons.tsx]: Maybe compare these to the icons used in the artifact list? I wonder if we can use the same ones in both locations?

File: client/webui/frontend/src/lib/components/chat/Citation.tsx (hunk 3 above)
Comment (JKaram) [reply client/webui/frontend/src/lib/components/chat/Citation.tsx:310]: We would be overriding almost all the styles. This is cleaner IMO

Will use theme.

**Comment by JKaram** [comment]
Re: feature flagging question for document sources

Document sources are gated on the backend via `project_indexing.enabled`. When disabled:

1. The BM25 index (`project_bm25_index.zip`) and converted text files are not copied to sessions
2. The LLM doesn't receive context about search capabilities
3. The `index_search` tool returns an error if called without an index

So if project indexing is disabled, the backend simply won't return `document_search` sources - the index won't exist in the session. No additional frontend gating needed.
